haskell-src/Concordium/Types/SeedState.hs

```diff
@@ -21,16 +61,43 @@ data SeedState = SeedState
       -- of blocks in the first 2/3 of the current epoch.
       updatedNonce :: !Hash
     }
-    deriving (Eq, Generic, Show)
+    deriving (Eq, Show)
+
+putSeedState :: Putter (SeedState ssv)
+putSeedState SeedState{..} = do
+    mapM_ put epochLength
+    put epoch
+    put currentLeadershipElectionNonce
+    put updatedNonce
 
-instance Serialize SeedState
+getSeedState :: SSeedStateVersion ssv -> Get (SeedState ssv)
+getSeedState ssv = do
+    epochLength <- conditionallyA (sSupportsEpochLength ssv) get
+    epoch <- get
+    currentLeadershipElectionNonce <- get
+    updatedNonce <- get
+    return SeedState{..}
+
+instance IsSeedStateVersion ssv => Serialize (SeedState ssv) where
+    put = putSeedState
+    get = getSeedState sing
 
 -- |Instantiate a seed state: leadership election nonce should be random, epoch length should be long, but not too long...
-initialSeedState :: LeadershipElectionNonce -> EpochLength -> SeedState
-initialSeedState nonce epochLength =
+initialSeedStateV0 :: LeadershipElectionNonce -> EpochLength -> SeedState 'SeedStateVersion0
+initialSeedStateV0 nonce theEpochLength =
+    SeedState
+        { epoch = 0,
+          epochLength = CTrue theEpochLength,
+          currentLeadershipElectionNonce = nonce,
+          updatedNonce = nonce
+        }
+
+-- |Instantiate a seed state: leadership election nonce should be random.
```

Comment:
It is unclear what the meaning of this comment is. What does random mean? And why is it important for this function.